netlify/cli
# fix: gracefully fail Edge Functions in case of deno binary issues

🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #4621 

Requires netlify/edge-bundler#42



`netlify dev` breaks in case one runs it on a platform (e.g linux aarch64) not yet officially supported by deno, to avoid this we need to gracefully fail Edge Functions making it possible to continue using `netlify dev`.

---

To test this:
- Clone `fix/add-error-state-to-after-download` branch of [edge-bundler](https://github.com/netlify/edge-bundler/tree/fix/add-error-state-to-after-download); run `npm install` and `npm run build` in edge-bundler to set it up; edit CLI's package.json to use the cloned version of edge-bundler eg `"@netlify/edge-bundler": "file:../edge-bundler" `
- Break the deno installation process by making the following changes in edge-bundler:
    - comment out this [block of code](https://github.com/netlify/edge-bundler/blob/main/src/bridge.ts#L164-L178) to force the download step to get called each time
    -  change [this line](https://github.com/netlify/edge-bundler/blob/main/src/bridge.ts#L67) to `if (downloadedVersion !== undefined) {` to cause the binary download step to throw an error.
    - change [this line](https://github.com/netlify/edge-bundler/blob/main/src/bridge.ts#L71) to `await this.writeVersionFile(downloadedVersion || '')` to get past TypeScript type check errors
    - run `npm run build` to make changes available CLI
- In a project use the local CLI to run the dev command e.g `~/netlify/cli/bin/run dev`
- It should surface the error without breaking rest of dev command process, test this by accessing a serverless function

Error string for review: `› There was a problem setting up the Edge Functions environment and it's unfortunately not possible to run Edge Functions from the CLI on this platform. More on supported platforms here: https://deno.land/manual/getting_started/installation. Error: <error.message from error object>`

Screenshot:

<img width="950" alt="Screenshot 2022-06-09 at 18 51 34" src="https://user-images.githubusercontent.com/16131515/172890751-8a310576-64bb-43fc-b212-a26742f4db48.png">


For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.

**A picture of a cute animal (not mandatory, but encouraged)**
![cute animal](https://res.cloudinary.com/cutelify/image/upload/c_fill,g_auto,q_auto:low,w_300,h_350/cat2_lkwgiz.jpg)
